.k8s/store-deployment.yml

```diff
@@ -0,0 +1,115 @@
+# Store Service deployment that creates a container for the app. It also mounts volumes for configuration and secret
+# files. The configuration files are created via manually added ConfigMaps called "store-config-map", "s3-config-map",
+# and transform_config.yml. The secret files are created manually with secret resources called "s3-secret-secret" and
+# "s3-access-secret".
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: store-deployment
+  labels:
+    app: store
+spec:
+  selector:
+    matchLabels:
+      app: store
+  template:
+    metadata:
+      labels:
+        app: store
+    spec:
+      containers:
+        - name: store
+          image:  registry.default.gsp.test:80/cnxta/ion-store
+          imagePullPolicy: Always
+          volumeMounts:
+            - name: store-configs
+              mountPath: /etc/store-configs
+              readOnly: true
+            - name: s3-configs
+              mountPath: /etc/s3-configs
+              readOnly: true
+            - name: transform-configs
+              mountPath: /etc/transform-configs
+              readOnly: true
+            - name: s3-access
+              mountPath: /etc/access-secret
+              readOnly: true
+            - name: s3-secret
+              mountPath: /etc/secret-secret
+              readOnly: true
+          ports:
+            - containerPort: 8080
+              protocol: TCP
+            - containerPort: 10051
+              protocol: TCP
+          args:
+            - "--s3.secret.file=/etc/secret-secret/s3_secret.sec"
+            - "--s3.access.file=/etc/access-secret/s3_access.sec"
+            - "--spring.config.additional-location=file:/etc/s3-configs/s3_config.yml"
+            - "--spring.config.additional-location=file:/etc/store-configs/store_config.yml"
+            - "--spring.config.additional-location=file:/etc/transform-configs/transform_config.yml"
+      volumes:
+        - name: store-configs
+          configMap:
+            name: store-config-map
+            defaultMode: 0755
+        - name: s3-configs
+          configMap:
+            name: s3-config-map
+            defaultMode: 0755
+        - name: transform-configs
+          configMap:
+            name: transform-config-map
+            defaultMode: 0755
+        - name: s3-secret
+          secret:
+            secretName: s3-secret-secret
+        - name: s3-access
+          secret:
+            secretName: s3-access-secret
+
+# Exposes the store container as a service that can be reached externally
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: ion-store
+  labels:
+    app: store
+spec:
+  selector:
+    app: store
+  ports:
+    - port: 8080
+      name: app
+    - port: 10051
+      name: debug
+
+# Adds Ingress for communicating externally and with other services in the namespace
```

Comment:
> with other services in the namespace

Ingress doesn't actually do this part. It's only responsible for exposing external to the cluster. A service is automatically communicable to other things within its namespace and within the cluster as long as its a `cluster ip`.